.github/workflows/ci.yml

```diff
@@ -22,3 +24,37 @@ jobs:
         run: |
           git-secrets --register-aws
           git-secrets --scan
+  doxygen:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          path: freertos
+      - name: Install Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7.10'
+      - name: Download tag dependency of coreMQTT-Agent
+        run: |
+          # We don't need to generate the coreMQTT docs, we only need the tag file.
+          # Therefore, we can just download it.
+          mkdir -p freertos/source/dependency/coreMQTT/docs/doxygen/output
+          wget -O freertos/source/dependency/coreMQTT/docs/doxygen/output/mqtt.tag \
+          "https://freertos.org/Documentation/api-ref/coreMQTT/docs/doxygen/output/mqtt.tag"
+      - name: Generate doxygen ZIP
+        uses: FreeRTOS/CI-CD-GitHub-Actions/doxygen@main
+        with:
+          path: ./freertos
+          # List of directories containing libraries whose doxygen output will be generated.
+          libs_parent_dir_path: FreeRTOS-Plus/Source,FreeRTOS-Plus/Source/AWS,FreeRTOS-Plus/Source/Application-Protocols
+          generate_zip: true
+      - name: Upload doxygen artifact if main branch
+        if: success() && github.ref == 'refs/heads/main'
+        uses: actions/upload-artifact@v2
+        with:
+          name: doxygen.zip
```

Comment:
I would prefer that there not be a zip file within a zip file here.

Reply:
The file is uploaded only as a single ZIP file, but GitHub only exports every artifact as an extra ZIP.

Reply:
This isn't really ideal. The output from the github action could really use some cleanup as well.

A few suggestions:
* Add a git short revision hash to the end of the directory that gets zipped
* Get rid of the intermediate "Source" directory inside FreeRTOS-Plus
* Collapse the docs/doxygen/output/html directories in each module.
* Should coreJSON and corePKCS11 be inside the Application-Protocols directory?

Reply:
The action is a generic utility which can be used from both C-SDK and FreeRTOS hub repositories. The folder structure does not affect the documentation as the libraries are individually parsed by the documentation before hosting on doxygen. 
At this point, updating the action for reaching the ideal goal may not be suitable. We can definitely consider these as improvements for the future. 

As part of the artifact uploading, I can add the git hash within the ZIP name.